catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/controller/SpriteController.java

```diff
@@ -106,10 +110,18 @@ public Sprite copyForCloneBrick(Sprite spriteToCopy) {
 
 		sprite.getSoundList().addAll(spriteToCopy.getSoundList());
 		sprite.getNfcTagList().addAll(spriteToCopy.getNfcTagList());
+		DeviceVariableAccessor accessor = new DeviceVariableAccessor(currentProject.getDirectory());
 
-		for (UserVariable userVariable : spriteToCopy.getUserVariables()) {
-			sprite.getUserVariables().add(new UserVariable(userVariable));
+		for (UserVariable originalVariable : spriteToCopy.getUserVariables()) {
+			if (checkDeviceVariableKeyEmpty(originalVariable)) {
```

Comment:
instead of setting those UUIDs every here and there on copy, clone, etc. why not generate them everytime? so that every user variable has one at all times?

Reply:
What do you mean with "generate them everytime"? 
First I thought it is not necessary for every UserVariable to have a valid key (!= null).
But if the clone call comes before WriteOnDevice, the Values of the Cloned and the Original Variable would be stored with different keys. So the key must be generated before. Another way would be to gernerate the key in the very beginning of the UserVariable, but I think there are way too many places where a Constructor is called (Clone, Backpack, Copy Sprite, almost every Brick).